### PR TITLE
Fix/sanitizerxn merge hs

### PR DESCRIPTION
**Pull request by bp-kelley**
Fixes incorrect exceptions regarding for "initMatchers" should be "initReactantMatchers"

Deprecated ChemDrawAdjustParams -> the ELN and sciquest integration doesn't follow this convention, so I suggest we make it more explicit "MatchOnlyAtRgroupsAdjustParams"

SanitizeRxn now can merge all hydrogens EXCEPT when a hydrogen is mapped to a non-hydrogen.

(Should this be a separate sanitize flag?  We can add this later if so)

Many tests added :)

### CODE REVIEW

File: Code/GraphMol/ChemReactions/SanitizeRxn.cpp [hunk 1]
```diff
@@ -324,15 +323,67 @@ void fixReactantTemplateAromaticity(ChemicalReaction &rxn) {
 }
 
 void fixHs(ChemicalReaction &rxn) {
+  {
+    //  if mapped Hydrogens in reactants area mapped to heavy atoms
+    //   keep mappings, in all other cases remove them.
+    //   this allows us to merge query hydrogens atoms 
+
+    std::map<int, bool> mappedToNonHeavyProductAtom;
```
Comment (greglandrum) [inline Code/GraphMol/ChemReactions/SanitizeRxn.cpp]: How about doing this with just a `dynamic_bitset`?

File: Code/GraphMol/ChemReactions/SanitizeRxn.h [hunk 2]
```diff
@@ -82,7 +82,22 @@ inline const MolOps::AdjustQueryParameters DefaultRxnAdjustParams() {
 }
 
 // Default adjustment parameters for ChemDraw style matching of reagents
+//  -- deprecated - renamed MatchOnlyAtRgroupsAdjustParams
+//  -- this doesn't match sciquest style searching
 inline const MolOps::AdjustQueryParameters ChemDrawRxnAdjustParams() {
```
Comment (greglandrum) [inline Code/GraphMol/ChemReactions/SanitizeRxn.h]: Missed this in the first pass: something should be added to the release notes for the deprecation and a github issue should probably be created

File: Code/GraphMol/ChemReactions/SanitizeRxn.cpp (hunk 1 above)
Comment (bp-kelley) [inline Code/GraphMol/ChemReactions/SanitizeRxn.cpp]: I just went to implement this and released we need to know the max atom map number or default the bitset to dynamic_bitset<> mappedNonHeavyProductAtoms(999).  I'm not sure doing two passes on the atoms is worth this optimization.

File: Code/GraphMol/ChemReactions/SanitizeRxn.h (hunk 2 above)
Comment (bp-kelley) [inline Code/GraphMol/ChemReactions/SanitizeRxn.h]: Added release note, if this pull is accepted I'll make an issue.

File: Code/GraphMol/ChemReactions/SanitizeRxn.cpp (hunk 1 above)
Comment (greglandrum) [inline Code/GraphMol/ChemReactions/SanitizeRxn.cpp]: of course you're right. I went through that too quickly and thought this was atom indexed.

**Pull request merged**